runtime/master/src/main/java/edu/snu/onyx/runtime/master/resource/ExecutorRepresenter.java

```diff
@@ -66,17 +70,22 @@ public void onExecutorFailed() {
   public void onTaskGroupScheduled(final ScheduledTaskGroup scheduledTaskGroup) {
     runningTaskGroups.add(scheduledTaskGroup.getTaskGroupId());
     failedTaskGroups.remove(scheduledTaskGroup.getTaskGroupId());
-
-    sendControlMessage(
-        ControlMessage.Message.newBuilder()
-            .setId(RuntimeIdGenerator.generateMessageId())
-            .setListenerId(MessageEnvironment.EXECUTOR_MESSAGE_LISTENER_ID)
-            .setType(ControlMessage.MessageType.ScheduleTaskGroup)
-            .setScheduleTaskGroupMsg(
-                ControlMessage.ScheduleTaskGroupMsg.newBuilder()
-                    .setTaskGroup(ByteString.copyFrom(SerializationUtils.serialize(scheduledTaskGroup)))
-                    .build())
-            .build());
+    
+    serializationExecutorService.submit(new Runnable() {
+      @Override
+      public void run() {
+        sendControlMessage(
```

Comment:
Move `SerializationUtils.serialize(scheduledTaskGroup)` to here to make serialization explicit?

```java
serialized = SerializationUtils.serialize(scheduledTaskGroup)
sendControlMessage(.... serialized ...)
```